src/test/kotlin/org/openrewrite/github/SetupJavaUpgradeJavaVersionTest.kt

```diff
@@ -0,0 +1,99 @@
+package org.openrewrite.github
+
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+import org.openrewrite.test.RecipeSpec
+import org.openrewrite.test.RewriteTest
+import org.openrewrite.yaml.Assertions
+import java.nio.file.Path
+
+class SetupJavaUpgradeJavaVersionTest : RewriteTest {
+    override fun defaults(spec: RecipeSpec) {
+        spec.recipe(SetupJavaUpgradeJavaVersion(17))
+    }
+
+    @Test
+    fun actionsSetupJavaVersionUpdates() = rewriteRun(
+        Assertions.yaml(
+            """
+            jobs:
+              build:
+                steps:
+                  - uses: actions/checkout@v2
+                    with:
+                      fetch-depth: 0
+                  - name: set-up-jdk-0
+                    uses: actions/setup-java@v2.3.0
+                    with:
+                      java-version: "11"
+                  - name: set-up-jdk-1
+                    uses: actions/setup-java@v2.3.0
+                    with:
+                      java-version: "11.0.17"
+                  - name: set-up-jdk-2
+                    uses: actions/setup-java@v2.3.0
+                    with:
+                      java-version: "17"
+                  - name: set-up-jdk-3
+                    uses: actions/setup-java@v2.3.0
+                    with:
+                      java-version: "17.0.4+11"
+                  - name: set-up-jdk-4
+                    uses: actions/setup-java@v2.3.0
+                    with:
+                      java-version: "18"
+                  - name: build
+                    run: ./gradlew build test
```

Comment:
When there are test failures I find it easier to debug when test cases are as specific as possible, covering exactly one scenario. Particularly since stepping through the visiting process or crafting an appropriate conditional breakpoint can be tedious.

I am also curious whether the `setup-java` action takes Java 8 versions as "8" or as "1.8". Before Java 9, java versions were listed as "1.x", and afterwards they dropped the 1. Tools that act on Java versions do not all agree on which version format should be used for old versions. There may well be some people who want to use this to update from java 8

Reply:
Based on the docs, I think `setup-java` only supports the post-1.x version naming: https://github.com/actions/setup-java#supported-version-syntax